pkg/controllers/cloud_config_sync_controller.go

```diff
@@ -16,18 +16,22 @@ import (
 	"sigs.k8s.io/controller-runtime/pkg/handler"
 	"sigs.k8s.io/controller-runtime/pkg/predicate"
 	"sigs.k8s.io/controller-runtime/pkg/source"
+
+	"github.com/openshift/cluster-cloud-controller-manager-operator/pkg/cloud"
 )
 
 const (
 	managedCloudConfigMapName = "kube-cloud-config"
 
-	defaultConfigKey = "cloud.conf"
-
 	// Controller conditions for the Cluster Operator resource
 	cloudConfigControllerAvailableCondition = "CloudConfigControllerAvailable"
 	cloudConfigControllerDegradedCondition  = "CloudConfigControllerDegraded"
 )
 
+// cloudConfigTransformer function transforms the source config map using the input infrastructure.config.openshift.io object.
+// only the data and binaryData field of the output ConfigMap will be respected by consumer of the transformer.
+type cloudConfigTransformer func(source *corev1.ConfigMap, infra *configv1.Infrastructure) (*corev1.ConfigMap, error)
```

Comment:
Is there somewhere that this signature can be made public rather than duplicating it between two packages?

Reply:
Sure. Will fix.